### PR TITLE
CLI: Try using enquirer to handle autocomplete prompts for changelog add

### PR DESCRIPTION
**Pull request by sdixon194**
An alternative to https://github.com/Automattic/jetpack/pull/34381

## Proposed changes:

Uses enquirer instead of inquirer to handle autocompletion when searching for changes.

This just handles the prompts for the `jetpack changelog add` command, and doesn't replace inquirer everywhere. If this is the direction we want to go later, we can look into that.
### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pf4qpu-9d-p2

## Does this pull request change what data or activity we track or use?

no
## Testing instructions:

Put the changelogger through its paces:

Try using `jetpack changelog add` and specify a project, or make some changes to projects and use `jetpack changelog add` to make sure everything works and the changelogger is put through its paces.

### CODE REVIEW

File: tools/cli/commands/changelog.js [hunk 1]
```diff
@@ -369,7 +370,8 @@ async function changelogAdd( argv ) {
 
 	for ( const proj of uniqueProjects ) {
 		console.log( chalk.green( `Running changelogger for ${ proj }!` ) );
-		const response = await promptChangelog( argv, proj, projectChangeTypes[ proj ] );
+		const projArray = [ proj ];
+		const response = await promptChangelog( argv, projArray, projectChangeTypes[ proj ] );
```
Comment (anomiex) [inline tools/cli/commands/changelog.js]: Does it not work to just do like this?
```suggestion
		const response = await promptChangelog( argv, [ proj ], projectChangeTypes[ proj ] );
```

File: tools/cli/commands/changelog.js [hunk 2]
```diff
@@ -734,72 +737,100 @@ async function promptChangelog( argv, needChangelog, types ) {
...
+	// Get the significance.
+	const { significance } = await prompt( {
+		type: 'autocomplete',
+		name: 'significance',
+		message: 'Significance of the change, in the style of semantic versioning.',
+		suggest: ( input, significanceChoices = [ 'patch', 'minor', 'major' ] ) => {
+			return Promise.resolve(
+				significanceChoices.filter( choice => choice.value.startsWith( input ) )
+			);
 		},
```
Comment (anomiex) [inline tools/cli/commands/changelog.js]: I think you overthought this.
```suggestion
		suggest: ( input, choices ) => choices.filter( choice => choice.value.startsWith( input ) ),
```

Although I note the highlighting enquirer applies is wonky, it doesn't match what's actually being searched. Maybe we should just turn it off.
```suggestion
		suggest: ( input, choices ) => choices.filter( choice => choice.value.startsWith( input ) ),
		highlight: v => v,
```

File: tools/cli/commands/changelog.js [hunk 3]
```diff
@@ -734,72 +737,100 @@ async function promptChangelog( argv, needChangelog, types ) {
...
+	} );
+
+	// Get the type of change.
+	const { type } = await prompt( {
+		type: 'autocomplete',
+		name: 'type',
+		message: 'Type of change.',
+		suggest: ( input, typeChoices = choices.map( choice => choice.value ) ) => {
+			return Promise.resolve( typeChoices.filter( choice => choice.value.startsWith( input ) ) );
 		},
```
Comment (anomiex) [inline tools/cli/commands/changelog.js]: Same as above
```suggestion
		suggest: ( input, choices ) => choices.filter( choice => choice.value.startsWith( input ) ),
		highlight: v => v,
```

File: tools/cli/commands/changelog.js (hunk 2 above)
Comment (sdixon194) [reply tools/cli/commands/changelog.js]: Ah yeah I did indeed.

**Comment by sdixon194** [comment]
Should be ready for review again, also moved where the changelog example text appears so it's right before the actual changelog entry prompt instead of as soon as you run the command now.

**Review by anomiex — APPROVED** [review]
Let's do it!

**Pull request merged**